crates/relayer/src/chain/namada.rs

```diff
@@ -0,0 +1,1235 @@
+        let config = CosmosSdkConfig {
+            proof_specs: Some(ibc_proof_specs::<Sha256Hasher>().into()),
+            ..config
+        };
+
+        Ok(Self {
+            config,
+            rpc_client,
+            wallet,
+            shielded_ctx,
+            native_token,
+            light_client,
+            rt,
+            keybase,
+            tx_monitor_cmd: None,
+        })
+    }
+
+    fn shutdown(self) -> Result<(), Error> {
+        Ok(())
+    }
+
+    fn health_check(&mut self) -> Result<HealthCheck, Error> {
+        self.rt
+            .block_on(SdkClient::health(&self.rpc_client))
+            .map_err(|e| {
+                Error::health_check_json_rpc(
+                    self.config.id.clone(),
+                    self.config.rpc_addr.to_string(),
+                    "/health".to_string(),
+                    e,
+                )
+            })?;
+
+        Ok(HealthCheck::Healthy)
+    }
+
+    fn subscribe(&mut self) -> Result<Subscription, Error> {
+        let tx_monitor_cmd = match &self.tx_monitor_cmd {
+            Some(tx_monitor_cmd) => tx_monitor_cmd,
+            None => {
+                let tx_monitor_cmd = self.init_event_source()?;
+                self.tx_monitor_cmd = Some(tx_monitor_cmd);
+                self.tx_monitor_cmd.as_ref().unwrap()
+            }
+        };
+
+        let subscription = tx_monitor_cmd.subscribe().map_err(Error::event_source)?;
+        Ok(subscription)
+    }
+
+    fn keybase(&self) -> &KeyRing<Self::SigningKeyPair> {
+        &self.keybase
+    }
+
+    fn keybase_mut(&mut self) -> &mut KeyRing<Self::SigningKeyPair> {
+        &mut self.keybase
+    }
+
+    fn get_key(&mut self) -> Result<Self::SigningKeyPair, Error> {
+        self.keybase
+            .get_key(&self.config.key_name)
+            .map_err(|e| Error::key_not_found(self.config.key_name.clone(), e))
+    }
+
+    fn get_signer(&self) -> Result<Signer, Error> {
+        let key = self
+            .keybase
+            .get_key(&self.config.key_name)
+            .map_err(|e| Error::key_not_found(self.config.key_name.clone(), e))?;
+        Ok(Signer::from_str(&key.account()).expect("The key name shouldn't be empty"))
+    }
+
+    fn ibc_version(&self) -> Result<Option<semver::Version>, Error> {
+        unimplemented!()
+    }
+
+    fn send_messages_and_wait_commit(
+        &mut self,
+        tracked_msgs: TrackedMsgs,
+    ) -> Result<Vec<IbcEventWithHeight>, Error> {
+        crate::time!("send_messages_and_wait_commit");
+
+        let proto_msgs = tracked_msgs.messages();
+        if proto_msgs.is_empty() {
+            return Ok(vec![]);
+        }
+        let mut tx_sync_results = vec![];
+        for msg in proto_msgs.iter() {
+            let response = self.send_tx(msg)?;
+
+            // Note: we don't have any height information in this case. This hack will fix itself
+            // once we remove the `ChainError` event (which is not actually an event)
```

Comment:
is this our hack or informal's hack?

Reply:
This is informal's hack.
https://github.com/heliaxdev/hermes/blob/21e7fd266f3e2440b726fc02882e146144ec333d/crates/relayer/src/chain/cosmos/batch.rs#L218C27-L219